sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/objects.scala

```diff
@@ -121,9 +121,7 @@ object EliminateMapObjects extends Rule[LogicalPlan] {
 object ObjectSerializerPruning extends Rule[LogicalPlan] {
 
   /**
-   * Collects all struct types from given data type object, recursively. Supports struct and array
-   * types for now.
-   * TODO(SPARK-26847): support map type.
+   * Collects all struct types from given data type object, recursively.
```

Comment:
What about `UserDefinedType`?

Reply:
This is for pruning object serializers. We don't use `UserDefinedType` in those serializers.

Reply:
If you don't mind, could you add `case UserDefinedType ` before `case _ =>` and add that comment please?

Reply:
Ping.

Reply:
sure. Will update later.